packages/main/src/TimePickerClock.ts

```diff
@@ -0,0 +1,1038 @@
+	 * @private
+	 */
+	@property({ type: Object })
+	_dimensionParameters!: TimePickerClockDimensions;
+
+	/**
+	 * Mousedown or Touchstart event flag.
+	 *
+	 * @type {Boolean}
+	 * @defaultvalue false
+	 * @private
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_mouseOrTouchDown!: boolean;
+
+	/**
+	 * Cancel Mouseout flag.
+	 *
+	 * @type {Boolean}
+	 * @defaultvalue false
+	 * @private
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_cancelTouchOut!: boolean;
+
+	/**
+	 * Visibility of '24' on Hours clock flag.
+	 *
+	 * @type {Boolean}
+	 * @defaultvalue false
+	 * @private
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_is24HoursVisible!: boolean;
+
+	/**
+	 * Calculated selected value of the clock during interactions.
+	 *
+	 * @type {Integer}
+	 * @defaultvalue -1
+	 * @private
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_selectedValue!: number;
+
+	/**
+	 * Selected value of the clock during interactions.
+	 *
+	 * @type {Integer}
+	 * @defaultvalue -1
+	 * @private
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_movSelectedValue!: number;
+
+	/**
+	 * Hovered value of the clock during interactions.
+	 *
+	 * @type {Integer}
+	 * @defaultvalue -1
+	 * @private
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_hoveredValue!: number;
+
+	/**
+	 * Previously hovered value of the clock during interactions.
+	 *
+	 * @type {Integer}
+	 * @defaultvalue -1
+	 * @private
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_prevHoveredValue!: number;
+
+	/**
+	 * Animation in progress flag.
+	 *
+	 * @type {Boolean}
+	 * @defaultvalue false
+	 * @private
+	 */
+	@property({ type: Boolean, noAttribute: true })
+	_animationInProgress!: boolean;
+
+	/**
+	 * Stores the ID of the long touch timeout.
+	 *
+	 * @type {Integer}
+	 * @defaultvalue -1
+	 * @private
+	 */
+	@property({ validator: Integer, defaultValue: -1, noAttribute: true })
+	_longTouchId!: number;
+
+	constructor() {
+		super();
+		// attach global event
+		document.addEventListener("mouseup", this._onMouseOutUp.bind(this), false);
+	}
+
+	get classes(): ClassMap {
+		return <ClassMap>{
+			clock: {
+				"ui5-tp-clock": true,
+				"ui5-tp-clock-inner": this.innerItems,
+				"ui5-tp-clock-active": this.active,
+			},
+		};
+	}
+
+	onBeforeRendering() {
+		this._prepareClockItems();
+		const value: number = this._fixReplacementValue(this.selectedValue);
+		this._updateSelectedValueObject(value);
+	}
+
+	/**
+	 * Returns the real value of the passed clock item, if the replacement must be done, returns the replaced value.
+	 *
+	 * @param {number} value The value of the clock item
+	 * @returns {number} The real/replaced value
+	 * @private
+	 */
+	_fixReplacementValue(value: number) {
+		let realValue = value;
+		const maxValue = this.itemMax * (this.innerItems ? 2 : 1);
+
+		if (!this.support2400)	{
+			if (realValue === 0) {
```

Comment:
Nope... Let me explain: 

The second IF does replacement of the value if (realValue === maxValue && this.lastItemReplacement !== -1).
but here we have 2 cases for the fist IF:

- if realValue is 0, it becomes maxValue
- but also realValue can be maxValue initially

so second IF should be checked always, not only as ELSE.
In addition, there cannot be composition of two IFs in one, like:

if ((realValue === maxValue || realValue === 0)  && this.lastItemReplacement !== -1) {
    realValue = this.lastItemReplacement;
}

because this.lastItemReplacement can be -1

So, to summarize, both IFs should be checked